rs-react/src/components/Header/Header.tsx

```diff
@@ -1,44 +1,54 @@
 import './Header.css';
-import { Component } from 'react';
+import { useState, useEffect, type Dispatch, type SetStateAction } from 'react';
 import { Button } from '../Button/Button';
+import { Link } from 'react-router-dom';
 
 type Props = {
   onSearchSubmit: (query: string) => void;
 };
 
-type State = {
-  inputValue: string;
-};
+function useLocalStorage(
+  key: string,
+  initialValue: string
+): [string, Dispatch<SetStateAction<string>>] {
+  const [value, setValue] = useState(() => {
+    return localStorage.getItem(key) || initialValue;
+  });
```

Comment:
```suggestion
function useLocalStorage(
  key: string,
  initialValue = ''
): [string, Dispatch<SetStateAction<string>>] {
  const [value, setValue] = useState(() => {
    return localStorage.getItem(key) || initialValue;
  });
```
Move shared code into separate files.

Reply:
Done